staging/src/k8s.io/apiserver/pkg/util/feature/testing/feature_gate_testing.go

```diff
@@ -18,13 +18,57 @@ package testing
 
 import (
 	"fmt"
+	"os"
+	"strings"
 	"testing"
 
 	"k8s.io/apiserver/pkg/util/feature"
 )
 
+// VerifyFeatureGatesUnchanged ensures the provided gate does not change any values when tests() are completed.
+// Intended to be placed into unit test packages that mess with feature gates.
+//
+// Example use:
+//
+// import (
+//   "testing"
+//
+//   utilfeature "k8s.io/apiserver/pkg/util/feature"
+//   utilfeaturetesting "k8s.io/apiserver/pkg/util/feature/testing"
+//   _ "k8s.io/kubernetes/pkg/features"
+// )
+//
+// func TestMain(m *testing.M) {
+//   utilfeaturetesting.VerifyFeatureGatesUnchanged(utilfeature.DefaultFeatureGate, m.Run)
+// }
+func VerifyFeatureGatesUnchanged(gate feature.FeatureGate, tests func() int) {
+	originalGates := gate.DeepCopy()
+	originalSet := fmt.Sprint(gate)
+
+	rc := tests()
+
+	finalSet := fmt.Sprint(gate)
+	if finalSet != originalSet {
```

Comment:
Follow up for someone - we should be able to read the feature gates instead of having ot parse from string.